tests/functional/openstackdataplanedeployment_controller_test.go

```diff
@@ -30,23 +38,116 @@ var _ = Describe("Dataplane Deployment Test", func() {
 			Namespace: namespace,
 			Name:      "dataplane-ansible-ssh-private-key-secret",
 		}
+		neutronOvnMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "neutron-ovn-metadata-agent-neutron-config",
+		}
+		novaNeutronMetadataSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-metadata-neutron-config",
+		}
+		novaCellComputeConfigSecretName = types.NamespacedName{
+			Namespace: namespace,
+			Name:      "nova-cell1-compute-config",
+		}
 		err := os.Setenv("OPERATOR_SERVICES", "../../config/services")
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	When("A dataplaneDeployment is created with matching NodeSet", func() {
 		BeforeEach(func() {
-			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNoNodeSetSpec()))
-			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 			CreateSSHSecret(dataplaneSSHSecretName)
+			CreateSSHSecret(neutronOvnMetadataSecretName)
+			CreateSSHSecret(novaNeutronMetadataSecretName)
+			CreateSSHSecret(novaCellComputeConfigSecretName)
+			DeferCleanup(th.DeleteInstance, CreateDataplaneNodeSet(dataplaneNodeSetName, DefaultDataPlaneNodeSetSpec()))
+			DeferCleanup(th.DeleteInstance, CreateDataplaneDeployment(dataplaneDeploymentName, DefaultDataPlaneDeploymentSpec()))
 		})
 
 		It("should have conditions set", func() {
```

Comment:
I would name this "should successfully deploy the requested services"

Reply:
I'd rather keep the name as it is. After all I'm only checking states of conditions, not actual deployment. When the test is expanded to make assertions on other things it may be time to revisit the issue.